pkg/virt-controller/watch/vm_test.go

```diff
@@ -5492,6 +5492,68 @@ var _ = Describe("VirtualMachine", func() {
 					Expect(err).To(Succeed())
 					Expect(vm.Spec.Preference.RevisionName).To(Equal(expectedPreferenceRevision.Name))
 				})
+
+				It("should apply preferences to hot plugged interface in VMI", func() {
+					testutils.UpdateFakeKubeVirtClusterConfig(kvInformer, &v1.KubeVirt{
+						Spec: v1.KubeVirtSpec{
+							Configuration: v1.KubeVirtConfiguration{
+								DeveloperConfiguration: &v1.DeveloperConfiguration{
+									FeatureGates: []string{
+										virtconfig.HotplugNetworkIfacesGate,
+									},
+								},
+							},
+						},
+					})
+					vm.Spec.Preference = &v1.PreferenceMatcher{
+						Name: preference.Name,
+						Kind: instancetypeapi.SingularPreferenceResourceName,
+					}
+					vm.Spec.Template.Spec.Networks = append(
+						vm.Spec.Template.Spec.Networks,
+						v1.Network{
+							Name: "old",
+						},
+					)
+					const originalInterfaceModel = "e1000"
+					vm.Spec.Template.Spec.Domain.Devices.Interfaces = append(
+						vm.Spec.Template.Spec.Domain.Devices.Interfaces,
+						v1.Interface{
+							Name: "old",
+							InterfaceBindingMethod: v1.InterfaceBindingMethod{
+								Bridge: &v1.InterfaceBridge{},
+							},
+							Model: originalInterfaceModel,
+						},
+					)
+					vm, err := virtFakeClient.KubevirtV1().VirtualMachines(vm.Namespace).Create(context.TODO(), vm, metav1.CreateOptions{})
```

Comment:
nit" `context.TODO()` is usually reserved for cases where one intends to replace it with a real context in the future. If there is no intention to do so, then `context.Background()` will be a good fit.
(seems it was used already in the VMI case below)